README.md

```diff
@@ -1,12 +1,264 @@
 [![Discord](https://img.shields.io/discord/478735028319158273.svg)](https://discord.gg/zAQBDEq)
 [![Discourse](https://img.shields.io/discourse/https/talk.fission.codes/topics)](https://talk.fission.codes)
 
-## 🚧 🚧 Under Construction 🚧 🚧
+Fission provides app hosting with user controlled data. We’re building a web native file system that combines files, encryption, and identity, like an open source iCloud.
+
+Get started making fission-enabled apps with the Fission SDK!
+
+## What you'll find here
+The Fission SDK offers tools for:
+- managing a user's web native **file system**
+- managing a user's **private keys**
+- **authenticating & authorizing** user actions
+
+You'll also find some helper functions for interacting with some of the building blocks:
+- [js-ipfs](https://github.com/ipfs/js-ipfs) for distributed file storage
+- [keystore-idb](https://github.com/fission-suite/keystore-idb) for key management, encryption & digital signatures
+
+
+# File System
+The Web Native File System (WNFS) is built on top of IPFS. It's structured and functions similarly to a Unix-style file system, with one notable exception: it's a Directed Acyclic Graph (DAG), meaning that a given child can have more than one parent (think symlinks but without the "sym").
+
+Each file system has a public tree and a private tree. All information (links, data, metadata, etc) in the private tree is encrypted. Decryption keys are stored in such a manner that access to a given folder grants access to all of its subfolders.
+
+## Basics
+WNFS exposes a familiar POSIX-style interface:
+- `ls`: list a directory
+- `mkdir`: create a directory
+- `add`: add a file
+- `cat`: retrieve a file
+
+## Versions
+Since the file system may evolve over time, a "version" is associated with each node in the file system (tracked with semver)
+
+Currently two versions exist:
+- `1.0.0`: file tree with metadata. Nodes in the file tree are structured as 2 layers where one layer contains "header" information (metadata, cache, etc), and the second layer contains data or links. 
+**This is the default version, use this unless you have a good reason not to**
+- `0.0.0`: bare file tree. The public tree is [ipfs dag-pg](https://github.com/ipld/js-ipld-dag-pb) nodes. The private tree is encrypted links with no associated metadata. These shuld really only be used for vanity links to be rendered by a gateway.
+
+## API
+
+### Config
+Each instantiation method takes an optional config. Below is the default config and descriptions of each value. 
+```ts
+const defaultConfig = {
+  keyName: 'filesystem-root', // the name of the key for the filesystem root as stored in IndexedDB
+  version: '1.0.0' // the version of the filesystem as discussed above
+}
+```
+--- 
+
+### Instantiation
+
+**empty**
+
+Creates a file system with an empty public tree & an empty private tree at the root
+
+Params: 
+- cfg: `FileSystemConfig` _optional_
+
+Example:
+```ts
+import FileSystem from 'fission-sdk/ffs'
```

Comment:
Should we rename `ffs` to just `fs`? 🤷

Reply:
Yes good call. had been meaning to do that